### PR TITLE
T05: Feature- User can get all the jobs

### PR DESCRIPTION
**Pull request by shinbe69**
All the users can use "/api/jobs" by GET method to get all the jobs. This API doesn't need authentication information.
1. Use "/api/jobs" API and remember to choose GET method.
2. Click send the request and if everything works properly, a list of jobs will be the result. That list contains all the jobs with all job property either.

Before:
image

After:
image

The link test server is the same: https://onboarding-job-board-api.herokuapp.com

### CODE REVIEW

File: src/api/job.js [hunk 1]
```diff
@@ -113,4 +113,22 @@ router.delete('/', async (request, response) => {
   }
 });
 
+router.get('/', async (request, response) => {
+  try {
+    const job = await Job.get();
```
Comment (tronghiendev) [inline src/api/job.js]: The result is an array, should be `jobs`

**Review by tronghiendev — CHANGES_REQUESTED** [review]

File: test/job.js [hunk 2]
```diff
@@ -169,15 +171,33 @@ describe('/CREATE, UPDATE, DELETE AND GET JOB', () => {
...
+    it('return status 200 and a list all of jobs in json file when get the jobs successful!', (done) => {
+      chai
+        .request(app())
+        .get('/api/jobs')
+        .end((err, response) => {
+          response.should.have.status(200);
+          response.body.should.have
+            .property('message')
+            .eql('Get all jobs successful, list of jobs is below!');
+          expect(response.body.jobs).to.be.a.jsonObj();
```
Comment (leofaba) [inline test/job.js]: If we don't check each job information details, how can we know the function is correct? In the other hands, `expect(response.body.jobs).to.be.a.jsonObj()` this expectation doesn't make sure the result is correct, right?

File: src/api/job.js [hunk 3]
```diff
@@ -113,4 +113,22 @@ router.delete('/', async (request, response) => {
...
+    }
+
+    return response.status(200).json({
+      message: 'Get all jobs successful, list of jobs is below!',
+      jobs
+    });
+  } catch (error) {
+    console.error(`getJob() >> Error: ${error.stack}`);
+
+    response.status(500).json({message: 'Get all of jobs failed!'});
```
Comment (leofaba) [inline src/api/job.js]: `catch (error)` we catch the error but not return to the client, how does the client know the error details if we only return the status code and a hard message like this `response.status(500).json({message: 'Get all of jobs failed!'});`?

**Review by leofaba — CHANGES_REQUESTED** [review]

File: src/api/job.js (hunk 3 above)
Comment (shinbe69) [reply src/api/job.js]: I thought that users should only know about whether it is succeeded or not, the person who needs more details is the developer. But we have to inform to normal users details of the error either?

File: test/job.js (hunk 2 above)
Comment (shinbe69) [reply test/job.js]: Fixed.

File: src/api/job.js (hunk 3 above)
Comment (shinbe69) [reply src/api/job.js]: Fixed.

**Pull request merged**